src/Equinox.MessageDb/MessageDb.fs

```diff
@@ -14,229 +16,127 @@ type EventBody = ReadOnlyMemory<byte>
-        | Some retryPolicy ->
-            let withLoggingContextWrapping count =
-                let log = if count = 1 then log else log |> prop contextLabel count
-                f log
-            retryPolicy withLoggingContextWrapping
-
-    /// NB Caveat emptor; this is subject to unlimited change without the major version changing - while the `dotnet-templates` repo will be kept in step, and
-    /// the ChangeLog will mention changes, it's critical to not assume that the presence or nature of these helpers be considered stable
-    module InternalMetrics =
-
-        module Stats =
-            let inline (|Stats|) ({ interval = i } : Measurement) = int64 i.ElapsedMilliseconds
-
-            let (|Read|ReadL|Write|Resync|Rollup|) = function
-                | Slice (Stats s) -> Read s
-                // slices are rolled up into batches so be sure not to double-count
-                | Batch (_, Stats s) -> Rollup s
-                | ReadLast (Stats s) -> ReadL s
-                | WriteSuccess (Stats s) -> Write s
-                | WriteConflict (Stats s) -> Resync s
-            type Counter =
-                { mutable count : int64; mutable ms : int64 }
-                static member Create() = { count = 0L; ms = 0L }
-                member x.Ingest(ms) =
-                    System.Threading.Interlocked.Increment(&x.count) |> ignore
-                    System.Threading.Interlocked.Add(&x.ms, ms) |> ignore
-            type LogSink() =
-                static let epoch = System.Diagnostics.Stopwatch.StartNew()
-                static member val Read = Counter.Create() with get, set
-                static member val ReadL = Counter.Create() with get, set
-                static member val Write = Counter.Create() with get, set
-                static member val Resync = Counter.Create() with get, set
-                static member Restart() =
-                    LogSink.Read <- Counter.Create()
-                    LogSink.ReadL <- Counter.Create()
-                    LogSink.Write <- Counter.Create()
-                    LogSink.Resync <- Counter.Create()
-                    let span = epoch.Elapsed
-                    epoch.Restart()
-                    span
-                interface Serilog.Core.ILogEventSink with
-                    member _.Emit logEvent = logEvent |> function
-                        | MetricEvent (Read stats) -> LogSink.Read.Ingest stats
-                        | MetricEvent (ReadL stats) -> LogSink.ReadL.Ingest stats
-                        | MetricEvent (Write stats) -> LogSink.Write.Ingest stats
-                        | MetricEvent (Resync stats) -> LogSink.Resync.Ingest stats
-                        | MetricEvent (Rollup _) -> ()
-                        | _ -> ()
-
-        /// Relies on feeding of metrics from Log through to Stats.LogSink
-        /// Use Stats.LogSink.Restart() to reset the start point (and stats) where relevant
-        let dump (log : ILogger) =
```

Comment:
example output of this in-context (TL;DR need to be able to produce similar stuff to that so one can analyze perf in the context of of an app that's saturating a rate limited store)

Every minute the scheduler dumps throughput info, and the equinox store module logs the summary of what it did (the same Metrics get fed to prometheus too)

<img width="859" alt="image" src="https://user-images.githubusercontent.com/206668/201494963-2012db5b-581b-47b5-82d6-107c7184501f.png">

Reply:
Not sure what to name it that's better but I'll grant you it's not something one would just discover! 

Yes, totally agree proper o11y stack is the true answer, but don't underestimate this poor man's aggregation mechanism can work very well for 90% of stuff - even when having the grafana charts rigged, for most ingestion/reaction jobs I tend to glance at the high level breakdown above (yes, despite it only having a pathetic average!) rather than the chart.